### PR TITLE
fix:(in-1921): added x-forwarded-host to condition

### PR DESCRIPTION
**Pull request by jeffpdotone**
Fix to add in the condition and see if there's an available x-forwarded-host headers

## Description
The changes will take in condition and see whether the x-forwarded-host is available or not fallback to another host property that can determine the domain/host it's coming from.

## Related Issue
Integrations we're not able to work correctly because of the missing host which is required by some of the APIs.

## Motivation and Context
This will prevent the integrations from not working because of the missing host.

## How Has This Been Tested?
Added unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

#### Changelog
- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [x] I have written test cases for my code
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

> I tested manually my code, and it works well with both:
- [ ] Default Theme
- [ ] Capybara Theme

#### Code standards
- [ ] My code follows the code style of this project.

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/core/src/utils/nuxt/_proxyUtils.ts [hunk 1]
```diff
@@ -43,12 +43,14 @@ export const getIntegrationConfig = (context: NuxtContext, configuration: any) =
...
 
+  const resolvedHost = context?.req?.headers?.['x-forwarded-host'] || context?.req?.headers?.host;
+
   return merge({
     axios: {
       baseURL,
       headers: {
         ...(cookie ? { cookie } : {}),
-        ...(context.req ? { Host: context.req.headers.host } : {})
+        ...(resolvedHost ? { Host: resolvedHost, 'x-forwarded-host': resolvedHost } : {})
```
Comment (WojtekTheWebDev) [inline packages/core/src/utils/nuxt/_proxyUtils.ts:53]: We don't need to add `'x-forwarded-host'` in here, cloud infra will do it on their side anyway. We should be just aware that when a traffic comes to the app from outside it comes as x-forwarded-host

**Review by WojtekTheWebDev — CHANGES_REQUESTED** [review]
Please change base branch to develop

File: packages/core/src/utils/nuxt/_proxyUtils.ts (hunk 1 above)
Comment (jeffpdotone) [reply packages/core/src/utils/nuxt/_proxyUtils.ts]: Removed thanks for clarifying that.

**Comment by Fifciu** [comment]
Already done here: https://github.com/vuestorefront/vue-storefront/pull/6818

**Comment by jeffpdotone** [comment]
> Already done here: #6818

Duplicates #6818

**Pull request closed without merging**